gobblin-runtime/src/main/java/gobblin/runtime/MysqlDatasetStateStore.java

```diff
@@ -51,9 +70,20 @@
 
   private static final Logger LOGGER = LoggerFactory.getLogger(MysqlDatasetStateStore.class);
 
+  private static final String SELECT_JOB_STATE_LATEST_BY_PREFIX_TEMPLATE =
+          "SELECT state, CASE WHEN locate(?, table_name) = 0 THEN '' "
```

Comment:
@htran1 I think I have a solution for this issue.  From what I can tell, the `getLatestDatasetStateFilePathsByUrns` is always passed the `jobName` as the `storeName`.  Additionally, we know that `tableName` will be in one of the following two formats:

1. `job_{{ NAME }}_{{ SEQUENCE }}.{{ EXTENSION }}`
2. `{{ URN }}-job_{{ NAME }}_{{ SEQUENCE }}.{{ EXTENSION }}`

What I propose is we look at the table name for the index of `-job_{{ NAME }}`.  In the case where there is no Urn, then it will be at `< 0`. In this case, the Urn will be set to `""`.  Otherwise, it will be `>= 0` and the Urn will be set to `tableName.substring(0, jobPrefixIndex)`.

This should correctly handle the case where the Urn contains characters such as a `-`.  Additionally, this logic should be able to be done in the Mysql query as well.